app/src/main/java/com/duckduckgo/app/tabs/ui/TabSwitcherActivity.kt

```diff
@@ -271,25 +272,18 @@ class TabSwitcherActivity : DuckDuckGoActivity(), TabSwitcherListener, Coroutine
     }
 
     override fun finish() {
-        clearObserversEarlyToStopViewUpdates()
         super.finish()
         overridePendingTransition(R.anim.slide_from_bottom, R.anim.tab_anim_fade_out)
     }
 
     override fun onDestroy() {
         super.onDestroy()
-        viewModel.deletableTabs.removeObservers(this)
         // we don't want to purge during device rotation
         if (isFinishing) {
             launch { viewModel.purgeDeletableTabs() }
         }
     }
 
-    private fun clearObserversEarlyToStopViewUpdates() {
```

Comment:
We don't need to manually cleanup anymore.

Reply:
Love this! 🤩